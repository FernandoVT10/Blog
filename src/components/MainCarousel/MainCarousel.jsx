import "./MainCarousel.scss";

export default () => {
    return (
        <div id="MainCarousel" className="carousel slide" data-ride="carousel">
            <ol className="carousel-indicators">
                <li data-target="#MainCarousel" data-slide-to="0" className="active"></li>
                <li data-target="#MainCarousel" data-slide-to="1"></li>
            </ol>

            <div className="carousel-inner">
                <div className="carousel-item carousel__item active">
                    <img
                    src="/img/carousel/carousel-1.jpg"
                    className="carousel__image"
                    alt="Carousel image" />
                </div>
                <div className="carousel-item">
                    <img
                    src="/img/carousel/carousel-2.jpg"
                    className="carousel__image"
                    alt="Carousel image" />
                </div>
            </div>
            
            <a
            className="carousel-control-prev carousel__control-button"
            href="#MainCarousel"
            role="button"
            data-slide="prev">
                <span className="carousel-control-prev-icon" aria-hidden="true"></span>
                <span className="sr-only">Previous</span>
            </a>

            <a
            className="carousel-control-next carousel__control-button"
            href="#MainCarousel" 
            ole="button"
            data-slide="next">
                <span className="carousel-control-next-icon" aria-hidden="true"></span>
                <span className="sr-only">Next</span>
            </a>
        </div>
    );
};