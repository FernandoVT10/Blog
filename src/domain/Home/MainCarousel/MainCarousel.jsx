import ApiController from "@/services/ApiController";

import { useEffect, useState } from "react";

import "./MainCarousel.scss";

const MainCarousel = () => {
    const [articles, setArticles] = useState([]);

    useEffect(() => {
        ApiController.get("articles?limit=4")
        .then(res => {
            if(res.data) {
                setArticles(res.data.articles);
            }
        });
    }, []);

    return (
        <div id="MainCarousel" className="carousel slide" data-ride="carousel">
            <ol className="carousel-indicators">
                {articles.map((_, index) => {
                    const indicatorClass = index ? "" : "active";

                    return (
                        <li
                        data-target="#MainCarousel"
                        data-slide-to={index}
                        className={indicatorClass}
                        key={index}></li>
                    );
                })}
            </ol>

            <div className="carousel-inner">
                {articles.map((article, index) => {
                    const itemClass = index ? "" : "active";

                    return (
                        <div
                        className={`carousel-item carousel__item ${itemClass}`}
                        key={index}>
                            <img
                            src={`/img/articles/${article.cover}`}
                            className="carousel__image"
                            alt={article.title}
                            loading="lazy" />
                            <div className="carousel-caption d-none d-md-block">
                                <h5 className="font-weight-bold">{article.title}</h5>
                            </div>
                        </div>
                    );
                })}
            </div>
            
            <a
            className="carousel-control-prev carousel__control-button"
            href="#MainCarousel"
            role="button"
            data-slide="prev">
                <span className="carousel-control-prev-icon" aria-hidden="true"></span>
                <span className="sr-only">Previous</span>
            </a>

            <a
            className="carousel-control-next carousel__control-button"
            href="#MainCarousel" 
            ole="button"
            data-slide="next">
                <span className="carousel-control-next-icon" aria-hidden="true"></span>
                <span className="sr-only">Next</span>
            </a>
        </div>
    );
}

export default MainCarousel;