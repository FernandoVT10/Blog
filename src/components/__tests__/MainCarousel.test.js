import MainCarousel from "../MainCarousel/";
import { create } from "react-test-renderer";

test("Check if <MainCarousel/> renders correctly", () => {
    const component = create(<MainCarousel/>);

    expect(component.toJSON()).toMatchSnapshot();
});