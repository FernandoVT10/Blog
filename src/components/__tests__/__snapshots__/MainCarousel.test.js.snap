// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if <MainCarousel/> renders correctly 1`] = `
<div
  className="carousel slide"
  data-ride="carousel"
  id="MainCarousel"
>
  <ol
    className="carousel-indicators"
  >
    <li
      className="active"
      data-slide-to="0"
      data-target="#MainCarousel"
    />
    <li
      data-slide-to="1"
      data-target="#MainCarousel"
    />
  </ol>
  <div
    className="carousel-inner"
  >
    <div
      className="carousel-item carousel__item active"
    >
      <img
        alt="Carousel image"
        className="carousel__image"
        src="/img/carousel/carousel-1.jpg"
      />
    </div>
    <div
      className="carousel-item"
    >
      <img
        alt="Carousel image"
        className="carousel__image"
        src="/img/carousel/carousel-2.jpg"
      />
    </div>
  </div>
  <a
    className="carousel-control-prev carousel__control-button"
    data-slide="prev"
    href="#MainCarousel"
    role="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
    <span
      className="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    className="carousel-control-next carousel__control-button"
    data-slide="next"
    href="#MainCarousel"
    ole="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
    <span
      className="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
